<?xml version="1.0" encoding="utf-8"?>
<robot name="turtlebot">
  <material name="Default-Material">
    <color rgba="1 1 1 1" />
  </material>
  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <mass value="5" />
      <inertia ixx="0.00999999977648258" ixy="0" ixz="0" iyy="0.00999999977648258" iyz="0" izz="0.5" />
    </inertial>
    <visual>
      <origin xyz="0 0 0.0308" />
      <geometry>
        <mesh filename="package://meshes/create_body.dae" scale="0.0254 0.0254 0.0254" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0.0308" />
      <geometry>
        <cylinder length="0.061163" radius="0.16495" />
      </geometry>
    </collision>
  </link>
  <link name="left_wheel_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <mass value="0.200000003" />
      <inertia ixx="0.00100000004749745" ixy="0" ixz="0" iyy="0.00100000004749745" iyz="0" izz="0.00100000004749745" />
    </inertial>
    <visual>
      <origin rpy="0 1.570796 -4.712485" />
      <geometry>
        <cylinder length="0.023" radius="0.033" />
      </geometry>
      <material name="Default-Material" />
    </visual>
    <collision>
      <geometry>
        <cylinder length="2" radius="0.5" />
      </geometry>
    </collision>
  </link>
  <link name="right_wheel_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <mass value="0.200000003" />
      <inertia ixx="0.00100000004749745" ixy="0" ixz="0" iyy="0.00100000004749745" iyz="0" izz="0.00100000004749745" />
    </inertial>
    <visual>
      <origin rpy="0 1.570796 -4.712485" />
      <geometry>
        <cylinder length="0.023" radius="0.033" />
      </geometry>
      <material name="Default-Material" />
    </visual>
    <collision>
      <origin rpy="-1.570796 0 0" />
      <geometry>
        <cylinder length="2" radius="0.5" />
      </geometry>
    </collision>
  </link>
  <link name="rear_wheel_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <mass value="0.0099999998" />
      <inertia ixx="9.99999974737875E-05" ixy="0" ixz="0" iyy="9.99999974737875E-05" iyz="0" izz="9.99999974737875E-05" />
    </inertial>
    <visual>
      <origin rpy="0 1.570796 -4.712485" />
      <geometry>
        <sphere radius="0.015" />
      </geometry>
      <material name="Default-Material" />
    </visual>
    <collision>
      <origin rpy="0 1.570796 -4.712485" />
      <geometry>
        <sphere radius="0.015" />
      </geometry>
    </collision>
  </link>
  <link name="front_wheel_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <mass value="0.0099999998" />
      <inertia ixx="0.00100000004749745" ixy="0" ixz="0" iyy="0.00100000004749745" iyz="0" izz="0.00100000004749745" />
    </inertial>
    <visual>
      <origin rpy="0 1.570796 -4.712485" />
      <geometry>
        <sphere radius="0.018" />
      </geometry>
      <material name="Default-Material" />
    </visual>
    <collision>
      <origin rpy="0 1.570796 -4.712485" />
      <geometry>
        <sphere radius="0.018" />
      </geometry>
    </collision>
  </link>
  <link name="plate_0_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0.050744 0 0.160223" />
      <mass value="1" />
      <inertia ixx="0.0254949387162924" ixy="0" ixz="0" iyy="0.0118522271513939" iyz="0" izz="0.0209144987165928" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://meshes/plate_0_logo.dae" scale="0.001 0.001 0.001" />
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://meshes/Mesh.stl" scale="1 1 1" />
      </geometry>
    </collision>
  </link>
  <link name="plate_1_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <mass value="0.1000000015" />
      <inertia ixx="0.00999999977648258" ixy="0" ixz="0" iyy="0.00999999977648258" iyz="0" izz="0.00999999977648258" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://meshes/plate_1_logo.dae" scale="0.001 0.001 0.001" />
      </geometry>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.008" radius="0.16" />
      </geometry>
    </collision>
  </link>
  <link name="plate_2_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <mass value="0.1000000015" />
      <inertia ixx="0.00100000004749745" ixy="0" ixz="0" iyy="0.00100000004749745" iyz="0" izz="0.00100000004749745" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://meshes/plate_1_logo.dae" scale="0.001 0.001 0.001" />
      </geometry>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.008" radius="0.16" />
      </geometry>
    </collision>
  </link>
  <link name="standoff_kinect_0_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <mass value="0.1000000015" />
      <inertia ixx="9.99999974737875E-05" ixy="0" ixz="0" iyy="9.99999974737875E-05" iyz="0" izz="9.99999974737875E-05" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://meshes/68-04556-RA_Kinect_Standoff_Assy.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0.043" />
      <geometry>
        <cylinder length="0.086" radius="0.0055" />
      </geometry>
    </collision>
  </link>
  <link name="camera_rgb_frame">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <mass value="0.5640000105" />
      <inertia ixx="0.00388124352321029" ixy="0" ixz="0" iyy="0.00049893994582817" iyz="0" izz="0.00387925724498928" />
    </inertial>
    <visual>
      <origin rpy="0 0 -4.712389" />
      <geometry>
        <mesh filename="package://meshes/kinect.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <geometry>
        <box size="0.07271 0.27794 0.073" />
      </geometry>
    </collision>
  </link>
  <link name="camera_rgb_optical_frame" />
  <link name="camera_depth_frame" />
  <link name="camera_depth_optical_frame" />
  <link name="standoff_kinect_1_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <mass value="0.1000000015" />
      <inertia ixx="9.99999974737875E-05" ixy="0" ixz="0" iyy="9.99999974737875E-05" iyz="0" izz="9.99999974737875E-05" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://meshes/68-04556-RA_Kinect_Standoff_Assy.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0.043" />
      <geometry>
        <cylinder length="0.086" radius="0.0055" />
      </geometry>
    </collision>
  </link>
  <link name="plate_3_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <mass value="0.1000000015" />
      <inertia ixx="9.99999974737875E-05" ixy="0" ixz="0" iyy="9.99999974737875E-05" iyz="0" izz="9.99999974737875E-05" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://meshes/plate_2_logo.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0.0144 0 0" />
      <geometry>
        <cylinder length="0.008" radius="0.16" />
      </geometry>
    </collision>
  </link>
  <link name="frontLaser">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <mass value="0.1299999952" />
      <inertia ixx="9.99999974737875E-05" ixy="0" ixz="0" iyy="9.99999974737875E-05" iyz="0" izz="9.99999974737875E-05" />
    </inertial>
  </link>
  <link name="backLaser">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <mass value="0.1299999952" />
      <inertia ixx="9.99999974737875E-05" ixy="0" ixz="0" iyy="9.99999974737875E-05" iyz="0" izz="9.99999974737875E-05" />
    </inertial>
  </link>
  <joint name="base_link_left_wheel_link_joint" type="continuous">
    <origin xyz="0 0.13 0.015" />
    <parent link="base_link" />
    <child link="left_wheel_link" />
    <axis xyz="0 1 0" />
    <dynamics damping="10" />
  </joint>

  <transmission name="left_wheel_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="base_link_left_wheel_link_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="left_motor">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <joint name="base_link_right_wheel_link_joint" type="continuous">
    <origin xyz="0 -0.13 0.015" />
    <parent link="base_link" />
    <child link="right_wheel_link" />
    <axis xyz="0 1 0" />
    <dynamics damping="10" />
  </joint>


  <transmission name="right_wheel_transmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="base_link_right_wheel_link_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="right_motor">
      <mechanicalReduction>1</mechanicalReduction>
      <hardwareInterface>VelocityJointInterface</hardwareInterface>
    </actuator>
  </transmission>

  <joint name="base_link_rear_wheel_link_joint" type="floating">
    <origin xyz="-0.13 0 0" />
    <parent link="base_link" />
    <child link="rear_wheel_link" />
  </joint>
  <joint name="base_link_front_wheel_link_joint" type="floating">
    <origin xyz="0.13 0 0" />
    <parent link="base_link" />
    <child link="front_wheel_link" />
  </joint>
  <joint name="base_link_plate_0_link_joint" type="fixed">
    <origin xyz="-0.04334 0 0.067757" />
    <parent link="base_link" />
    <child link="plate_0_link" />
  </joint>
  <joint name="plate_0_link_plate_1_link_joint" type="fixed">
    <origin xyz="0.04068 0 0.05715" />
    <parent link="plate_0_link" />
    <child link="plate_1_link" />
  </joint>
  <joint name="plate_1_link_plate_2_link_joint" type="fixed">
    <origin xyz="0 0 0.057201" />
    <parent link="plate_1_link" />
    <child link="plate_2_link" />
  </joint>
  <joint name="plate_2_link_standoff_kinect_0_link_joint" type="fixed">
    <origin xyz="-0.102438 0.098 0.0032" />
    <parent link="plate_2_link" />
    <child link="standoff_kinect_0_link" />
  </joint>
  <joint name="standoff_kinect_0_link_camera_rgb_frame_joint" type="fixed">
    <origin xyz="-0.012902 -0.098 0.085692" />
    <parent link="standoff_kinect_0_link" />
    <child link="camera_rgb_frame" />
  </joint>
  <joint name="camera_rgb_frame_camera_rgb_optical_frame_joint" type="fixed">
    <origin rpy="-1.570796 0 -1.570796" xyz="0.018098 -0.1105 0.101692" />
    <parent link="camera_rgb_frame" />
    <child link="camera_rgb_optical_frame" />
  </joint>
  <joint name="camera_rgb_frame_camera_depth_frame_joint" type="fixed">
    <origin xyz="0.018098 -0.0855 0.101692" />
    <parent link="camera_rgb_frame" />
    <child link="camera_depth_frame" />
  </joint>
  <joint name="camera_depth_frame_camera_depth_optical_frame_joint" type="fixed">
    <origin rpy="-1.570796 0 -1.570796" />
    <parent link="camera_depth_frame" />
    <child link="camera_depth_optical_frame" />
  </joint>
  <joint name="plate_2_link_standoff_kinect_1_link_joint" type="fixed">
    <origin xyz="-0.102438 -0.098 0.0032" />
    <parent link="plate_2_link" />
    <child link="standoff_kinect_1_link" />
  </joint>
  <joint name="plate_2_link_plate_3_link_joint" type="fixed">
    <origin xyz="-0.01316 0 0.20635" />
    <parent link="plate_2_link" />
    <child link="plate_3_link" />
  </joint>
  <joint name="plate_3_link_frontLaser_joint" type="fixed">
    <origin xyz="0.1 0 0.006" />
    <parent link="plate_3_link" />
    <child link="frontLaser" />
  </joint>
  <joint name="plate_3_link_backLaser_joint" type="fixed">
    <origin rpy="0 0 -3.141593" xyz="-0.1 0 0.006" />
    <parent link="plate_3_link" />
    <child link="backLaser" />
  </joint>
</robot>